peps/pep-0765.rst

```diff
@@ -0,0 +1,169 @@
+PEP: 765
+Title: Disallow return/break/continue that exit a finally block
+Author: Irit Katriel <irit@python.org>, Alyssa Coghlan <ncoghlan@gmail.com>
+Discussions-To: https://discuss.python.org/t/an-analysis-of-return-in-finally-in-the-wild/70633
+Status: Draft
+Type: Standards Track
+Created: 15-Nov-2024
+Python-Version: 3.14
+Post-History: 09-Nov-2024
+
+.. highlight:: rst
+
+Abstract
+========
+
+This PEP proposes to withdraw support for ``return``, ``break`` and
+``continue`` statements that break out of a ``finally`` block.
+This was proposed in the past by :pep:`601`. The current PEP
+is based on empirical evidence regarding the cost/benefit of
+this change, which did not exist at the time that :pep:`601`
+was rejected. It also proposes a slightly different solution
+than that which was proposed by :pep:`601`.
+
+Motivation
+==========
+
+The semantics of ``return``, ``break`` and ``continue`` in a
+finally block are surprising for many developers.
+The documentation [1]_ mentions that
+
+1. If the ``finally`` clause executes a ``break``, ``continue``
+or ``return`` statement, exceptions are not re-raised.
+2. If a ``finally`` clause includes a ``return`` statement, the
+returned value will be the one from the ``finally`` clause’s
+``return`` statement, not the value from the ``try`` clause’s
+``return`` statement.
```

Comment:
Let's render these as separate items rather than a single paragraph. And use bullets because there's no inherent order for these statements:

```suggestion
- If the ``finally`` clause executes a ``break``, ``continue``
  or ``return`` statement, exceptions are not re-raised.

- If a ``finally`` clause includes a ``return`` statement, the
  returned value will be the one from the ``finally`` clause’s
  ``return`` statement, not the value from the ``try`` clause’s
  ``return`` statement.
```